app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,30 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: post } }
+    end
+  end
+
+  def create
+    post = Post.new(params.require(:post).permit(:title, :text))
+    user = User.find(params[:user_id])
+    post.author = user
+    post.comments_counter = 0
+    post.likes_counter = 0
+    respond_to do |format|
+      format.html do
+        if post.save && user.update_attribute(:posts_counter, user.posts_counter + 1)
+          flash[:success] = 'post saved successfully'
+          redirect_to posts_index_url(user.id)
```

Comment:
- [ ] <b>OPTIONAL:</b> You have create a function called `post_update_counter` in Post model, instead of using `user.update_attribute(:posts_counter, user.posts_counter + 1)` to update post count. you can use after_save feature. in you post model, add the following line on top:

```ruby
after_save :post_update_counter